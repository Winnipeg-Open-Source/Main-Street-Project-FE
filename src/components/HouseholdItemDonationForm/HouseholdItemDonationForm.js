import React, { memo } from 'react';
import PropTypes from 'prop-types';
import Input from 'components/Input';
import LineItemForm from 'components/LineItemForm';
import TextArea from 'components/TextArea';

function HouseholdItemDonationForm ({
    name,
    description,
    quantityName,
    locations,
    lineItems,
    notes,
    onChange,
    onAddLineItemClick,
    onItemQuantityChange,
}) {
    return (
        <>
            <Input
                id='name'
                name='name'
                placeholder='Name'
                value={name}
                onChange={onChange}
            />
            <Input
                id='description'
                name='description'
                placeholder='Description'
                value={description}
                onChange={onChange}
            />
            <Input
                id='quantity-name'
                name='quantityName'
                placeholder='Quantity Name'
                value={quantityName}
                onChange={onChange}
            />
            <LineItemForm
                locations={locations}
                lineItems={lineItems}
                onAddClick={onAddLineItemClick}
                onItemQuantityChange={onItemQuantityChange}
            />
            <TextArea
                id='notes'
                name='notes'
                label='Notes'
                value={notes}
                onChange={onChange}
            />
        </>
    );
}

HouseholdItemDonationForm.displayName = 'HouseholdItemDonationForm';

HouseholdItemDonationForm.propTypes = {
    name: PropTypes.string,
    description: PropTypes.string,
    quantityName: PropTypes.string,
    locations: PropTypes.array,
    lineItems: PropTypes.array,
    notes: PropTypes.string,
    onChange: PropTypes.func,
    onAddLineItemClick: PropTypes.func,
    onItemQuantityChange: PropTypes.func,
};

export default memo(HouseholdItemDonationForm);
