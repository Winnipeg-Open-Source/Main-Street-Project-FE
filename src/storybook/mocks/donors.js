export const noDonors = {
    isLoading: false,
    donors: [],
};

export const singleDonor = {
    isLoading: false,
    donors: [
        {
            id: 1,
            name: 'Costco',
            address: '22 Road Lane',
            city: 'Winnipeg, MB',
            email: 'costco@costco.com',
            phoneNumber: '204-555-5555',
            lastDonationDate: '2019 December 20',
            donations: [
                {
                    donationDate: '2019 December 20',
                },
                {
                    donationDate: '2019 November 20',
                },
            ],
        },
    ],
};

export const multipleDonors = {
    isLoading: false,
    donors: [
        {
            id: 1,
            name: 'Costco',
            lastDonationDate: '2019 December 20',
        },
        {
            id: 2,
            name: 'Sobeys',
            lastDonationDate: '2019 December 20',
        },
        {
            id: 3,
            name: 'Jane Doe',
            lastDonationDate: '2019 December 20',
        },
    ],
};
