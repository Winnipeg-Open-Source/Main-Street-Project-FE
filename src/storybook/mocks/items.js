export const mockItemRequisition = [
    {
        id: 1,
        name: 'Paper Towel',
        quantity: 5,
        quantityName: 'Rolls',
    },
    {
        id: 2,
        name: 'Blueberries',
        quantity: 10,
        quantityName: 'Packages',
    },
    {
        id: 3,
        name: 'Steaks',
        quantity: 2,
    },
];