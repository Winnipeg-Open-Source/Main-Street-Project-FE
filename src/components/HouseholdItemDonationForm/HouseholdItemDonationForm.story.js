import React from 'react';
import { action } from '@storybook/addon-actions';
import Form from 'components/Form';
import HouseholdItemDonationForm from './HouseholdItemDonationForm';
import { mockLocations } from 'tests/mocks/locations';
import { mockLineItems } from 'tests/mocks/items';

const props = {
    name: 'Toilet Paper',
    description: 'This toilet paper is 10-ply',
    quantityName: 'Rolls',
    locations: mockLocations,
    lineItems: mockLineItems,
    notes: 'A bunch of notes',
    onChange: action('on change'),
    onAddLineItemClick: action('on line item clicked'),
    onItemQuantityChange: action('on line item quantity changed'),
};

export const EmptyForm = () => (
    <Form>
        <HouseholdItemDonationForm />
    </Form>
);

export const CompletedForm = () => (
    <Form>
        <HouseholdItemDonationForm {...props} />
    </Form>
);

export default {
    title: 'Forms/HouseholdItemDonationForm',
    component: HouseholdItemDonationForm,
};
