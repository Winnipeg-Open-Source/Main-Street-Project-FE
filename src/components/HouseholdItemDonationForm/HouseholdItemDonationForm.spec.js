import React from 'react';
import { render } from '@testing-library/react';
import HouseholdItemDonationForm from './HouseholdItemDonationForm';
import { mockHouseholdItemDonationProps } from 'tests/mocks/MockDonationFormProps';

describe ('HouseholdItemDonationForm', () => {
    it ('renders without crashing', () => {
        const { asFragment } = render(<HouseholdItemDonationForm />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('has placeholders when empty', async () => {
        const { getByLabelText, getByPlaceholderText } = render(<HouseholdItemDonationForm />);

        const name = await getByPlaceholderText('Name');
        const description = await getByPlaceholderText('Description');
        const quantity = await getByPlaceholderText('Quantity');
        const notes = await getByLabelText('Notes');

        expect(name.value).toBe('');
        expect(description.value).toBe('');
        expect(quantity.value).toBe('');
        expect(notes.value).toBe('');
    });

    it ('has all fields used', async () => {
        const { getByLabelText, getByPlaceholderText } = render(<HouseholdItemDonationForm {...mockHouseholdItemDonationProps} />);

        const name = await getByPlaceholderText('Name');
        const description = await getByPlaceholderText('Description');
        const quantity = await getByPlaceholderText('Quantity');
        const notes = await getByLabelText('Notes');

        expect(name.value).toBe('Toilet Paper');
        expect(description.value).toBe('This toilet paper is 10-ply');
        expect(quantity.value).toBe('5');
        expect(notes.value).toBe('A bunch of notes');
    });
});
