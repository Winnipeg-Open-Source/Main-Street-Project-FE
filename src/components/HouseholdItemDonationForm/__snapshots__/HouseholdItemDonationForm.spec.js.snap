// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseholdItemDonationForm renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-pkfPC haHNKR sc-psDhf iAENMo"
  >
    <input
      class="sc-qcrrk sc-qQxFs iUvCRq"
      id="name"
      name="name"
      placeholder="Name"
      value=""
    />
    <input
      class="sc-qcrrk sc-qQxFs iUvCRq"
      id="description"
      name="description"
      placeholder="Description"
      value=""
    />
    <input
      class="sc-qcrrk sc-qQxFs iUvCRq"
      id="quantity-name"
      name="quantityName"
      placeholder="Quantity Name"
      value=""
    />
    <div
      class="sc-AxirZ sc-AxhCb sc-pbvYO lbwXdz"
      width="1"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="downshift-0-label"
        role="combobox"
        style="position: relative; width: 100%;"
      >
        <input
          aria-autocomplete="list"
          aria-labelledby="downshift-0-label"
          autocomplete="off"
          class="sc-qcrrk gTabPy sc-pJipy guPmcd"
          font-size="2,,1"
          id="downshift-0-input"
          placeholder="Location Search"
          value=""
          width="1"
        />
        <div
          aria-labelledby="downshift-0-label"
          class="sc-AxirZ sc-pCPXO sc-qYUWV iPxrxw"
          height="256px"
          id="downshift-0-menu"
          role="listbox"
        />
      </div>
    </div>
    <label
      class="sc-pdKru dQCRbh"
      color="border.light"
      font-size="10px"
      font-weight="bold"
    >
      Notes
      <textarea
        class="sc-pAYXY sc-prrfo jgdyQH"
        id="notes"
        name="notes"
      />
    </label>
  </div>
</DocumentFragment>
`;
